component/rbac_testing.jsonnet

```diff
@@ -0,0 +1,119 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.appcat;
+
+local e2eNs = kube.Namespace('appcat-e2e') + {
+  metadata+: {
+    labels+: {
+      'appuio.io/organization': 'vshn',
+    },
+  },
+};
+
+local e2eSA = kube.ServiceAccount('github-ci') + {
+  metadata+: {
+    namespace: 'appcat-e2e',
+    labels: {
+      'appuio.io/organization': 'vshn',
+    },
+  },
+};
+
+local e2eRoleBinding = kube.RoleBinding('appcat-e2e') + {
+  metadata+: {
+    namespace: 'appcat-e2e',
+  },
+  roleRef_: kube.ClusterRole('admin'),
+  subjects_: [ e2eSA ],
+};
+
+local e2eClusterRole = kube.ClusterRole('appcat:e2e') + {
+  rules: [
+    {
+      apiGroups: [ 'networking.k8s.io' ],
+      resources: [ 'ingresses' ],
+      verbs: [ 'get', 'list' ],
+    },
+    {
+      apiGroups: [ '' ],
+      resources: [ 'namespaces' ],
+      verbs: [ 'delete' ],
+    },
+    {
+      apiGroups: [ 'batch' ],
+      resources: [ 'jobs', 'cronjobs', 'jobs/finalizers', 'cronjobs/finalizers' ],
+      verbs: [ 'get', 'list', 'create', 'delete', 'update', 'watch' ],
+    },
+  ],
+};
+
+local e2eClusterRoleBinding = kube.ClusterRoleBinding('appcat:e2e') {
+  roleRef_: e2eClusterRole,
+  subjects_: [ e2eSA ],
+};
+
+local e2eSAToken = kube.Secret('github-ci-secret') + {
+  metadata+: {
+    namespace: 'appcat-e2e',
+    annotations+: {
+      'kubernetes.io/service-account.name': 'github-ci',
+      'argocd.argoproj.io/compare-options': 'IgnoreExtraneous',
+    },
+  },
+  type: 'kubernetes.io/service-account-token',
+};
+
+local diffSA = kube.ServiceAccount('crossplane-diff') + {
+  metadata+: {
+    namespace: params.namespace,
+    labels: {
+      'appuio.io/organization': 'vshn',
+    },
+  },
+};
+
+local diffClusterRoleBindingManagedResources = kube.ClusterRoleBinding('appcat:render-diff:managed-resources') {
+  roleRef: {
+    // the clusterrole `crossplane-view` is managed by crossplane and has view
+    // permissions on all installed managed resources.
+    name: 'crossplane-view',
+    apiGroup: 'rbac.authorization.k8s.io',
+    kind: 'ClusterRole',
+  },
+  subjects_: [ diffSA ],
+};
+
+local diffClusterRole = kube.ClusterRole('appcat:render-diff:functions') + {
+  rules: [
+    {
+      apiGroups: [ 'pkg.crossplane.io' ],
+      resources: [ 'functions' ],
+      verbs: [ 'get', 'list' ],
+    },
+  ],
+};
```

Comment:
shouldn't be there composites as well?

Reply:
The composites are also part of the `crossplane-view` clusterrole:
```
ka get clusterrole crossplane-view -oyaml | grep xvshn
  - xvshnkeycloaks
  - xvshnkeycloaks/status
  - xvshnmariadbs
  - xvshnmariadbs/status
  - xvshnnextclouds
  - xvshnnextclouds/status
  - xvshnpostgresqls
  - xvshnpostgresqls/status
  - xvshnredis
  - xvshnredis/status
```